src/main/java/com/google/sps/servlets/ImageHandlerServlet.java

```diff
@@ -31,108 +32,105 @@
-    } catch (Exception e) {
-    }
-    return entity;
-  }
+  private Entity player;
+  private static final DatastoreService DATASTORE = DatastoreServiceFactory.getDatastoreService();
+  private boolean isLoggedIn = UserServiceFactory.getUserService().isUserLoggedIn();
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String imageUrl;
-    try {
-      if (!isLoggedIn) {
-        response.sendRedirect(LOGIN_REDIRECT_PARAMETER);
-      } else {
-        // Get the displayName entered by the user.
-        String displayName = request.getParameter(DISPLAY_NAME_PARAMETER);
-        // Get the imageName to check if a file was uploaded.
-        String imageName = request.getParameter(IMAGE_PARAMETER);
-        if (imageName == null) { // Default image path
-          BlobKey defaultBlobKey = blobstoreService.createGsBlobKey(DEFAULT_IMAGE_GS_LOCATION);
-          imageUrl = getFileUrl(defaultBlobKey);
-        } else {
-          // Get the URL of the image that the user uploaded to Blobstore.
-          BlobKey uploadedBlobKey = createUploadedBlobKey(request, IMAGE_PARAMETER);
-          imageUrl = getFileUrl(uploadedBlobKey);
-        }
-        // Assign imageUrl to current player
-        PlayerDatabase.setEntityImageID(imageUrl);
+    String displayName = request.getParameter(MESSAGE_PARAMETER);
 
-        // Assign displayName to current player
-        PlayerDatabase.setEntityDisplayName(displayName);
-      }
-      response.sendRedirect(UPLOADED_REDIRECT_PARAMETER);
-    } catch (Exception e) {
+    UserService userService = UserServiceFactory.getUserService();
+    if (!isLoggedIn) {
+      response.sendRedirect(LOGIN_REDIRECT_PARAMETER);
+      return;
+    }
+    String imageBlobKeyString = createUploadedBlobKey(request, IMAGE_PARAMETER);
+    if (imageBlobKeyString.equals("false")) {
+      response.sendRedirect(CHARACTER_DESIGN_REDIRECT_PARAMETER);
+      return;
     }
+
+    // create and store store current player
+    DATASTORE.put(newPlayerEntity(imageBlobKeyString, displayName));
+    response.sendRedirect(UPLOADED_REDIRECT_PARAMETER);
   }
 
-  // Gets the URL of the uploaded file; null if no file was uploaded
-  private BlobKey createUploadedBlobKey(HttpServletRequest request, String formInputElementName) {
+  private Entity newPlayerEntity(String imageBlobKeyString, String displayName) {
+    player = new Entity(PLAYER_QUERY_PARAMETER);
+    player.setProperty(EMAIL_PARAMETER, USER.getEmail());
+    player.setProperty(IMAGE_ID_PARAMETER, imageBlobKeyString);
+    player.setProperty(DISPLAY_NAME_PARAMETER, displayName);
+    player.setProperty(ID_PARAMETER, USER.getUserId());
+    return player;
+  }
+
+  private String createUploadedBlobKey(HttpServletRequest request, String formInputElementName) {
     Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
     List<BlobKey> blobKeys = blobs.get(formInputElementName);
```

Comment:
How does this key generation work? Are two users guarantee'd different keys even if their image has the same name? If not, there's an issue where one user can delete another users content.